### ISSUE TITLE
Precompile that exposes execution-node's API

### ISSUE THREAD

**Issue by halo3mic** [flashbots/suave-geth#208]
### Describe the feature you would like

A precompile that exposes execution-node's API. For starters `eth` namespace would be enough.

Afaik now one needs to call external node to get the latest block or account balance, this seems unnecessary as there is a local node running alongside SUAVE. 
It is true that some values could be obtained by `eth_call` to a contract exposing block API ([like the Maker's MulticallV2](https://github.com/makerdao/multicall/blob/master/src/Multicall2.sol)), but this again seems more convoluted that it should be.

**Comment by lthibault** [comment]
In principle, could this be achieved using SUAVE's HTTP precompile?  It seems like it would be a straightforward matter of providing you with the execution node's URL, and then leaving it up to you to perform JSON-RPC requests.

(cc @ferranbt)

**Comment by halo3mic** [comment]
I guess this would work if you know the port of the execution node. But not sure how cautious one should be with exposing the local node.

**Comment by lthibault** [comment]
Agreed.  I'm going to bring this up with the team and get back to you.

**Comment by lthibault** [comment]
@halo3mic so the issue is that we don't want to leak the node's IP:PORT tuple, as it exposes us to DoS.  However, I believe [this recently-added feature](https://github.com/flashbots/suave-geth/pull/196) would satisfy your requirement.

Instead of
```
string memory httpAddr = Suvve.resolveDns("goerli")
Suave.httpRequest({.... addr: httpAddr})
```
You would do something like  `Suave.httpRequest({... addr: "goerli"})`.

**Comment by lthibault** [comment]
>Do you think this still exposes the execution node to ddos - you can do CCR using this new feature without any cost (if MEVM execution fails no gas is paid)?

I believe you are correct, though this is on our list of things to figure out "in the future".  Off-the-cuff, gating this through a CCR allows us to do things like load balancing, or possibly introducing some sort of gas tax.  TBD.

Glad this fixes the issue! 🙂
